src/js/api/Setup.js

```diff
@@ -43,15 +43,43 @@ const Setup = function(_model) {
                 throw error;
             };
             return pluginsPromise.then(throwError).catch(throwError);
-        });
+        }).then(allPromises => setupResult(allPromises));
     };
 
     this.destroy = function() {
         clearTimeout(_setupFailureTimeout);
         _model.set('_destroyed', true);
         _model = null;
     };
-
 };
 
+/**
+ * @typedef { object } SetupResult
+ * @property { object } core
+ * @property {Array<PlayerError>} warnings
+ */
+
+/**
+ *
+ * @param {Array<Promise>} allPromises - An array of promise resolutions or rejections
+ * @returns {SetupResult} setupResult
+ */
+export function setupResult(allPromises) {
+    if (!allPromises || !allPromises.length) {
+        return {
+            core: null,
+            warnings: []
+        };
+    }
+
+    const warnings = allPromises
+        .reduce((acc, val) => acc.concat(val), []) // Flattens the sub-arrays of allPromises into a single array
+        .filter(p => p && p.code);
```

Comment:
p is promise ? is there a more descriptive word we could use ?

Reply:
It's the result of each promise. If it has a code, it's an PlayerError warning result (setup errors throw so they skip this and go to the catch handler).